src/puppetlabs/trapperkeeper/authorization/acl.clj

```diff
@@ -44,57 +46,58 @@
 
 ;; ACE creation
 
-(defn munge-name
-  [pattern]
-  (-> pattern (str/lower-case) (str/split #"\.") reverse vec))
+(schema/defn ^:private split-domain :- [String]
+  "Given a domain string, split it on '.' and reverse it. For examples,
+  'sylvia.plath.net' becomes ['net' 'plath' 'sylvia']. This is used for domain
+  matching."
+  [pattern :- String]
```

Comment:
I hate this name. I was putting off thinking of something new until I got back to TK-293 proper but was considering something vague like "value" :/